### ISSUE TITLE
Question about pre-train model

### ISSUE THREAD

**Issue by LiXm1002** [VisualComputingInstitute/TrackR-CNN#44]
I am confused about the weight in which part of network will be load when I use the pre-train model named trackrcnn_init. And when training, the weight of whole network will change gradually or just the part which don't load pre-train model?
Thanks for answering question.

**Comment by ahnonay** [comment]
Hey! The trackrcnn_init model corresponds to Mask R-CNN pretrained on COCO and Mapillary. So all weights are initialized except the ones of the association head. We then finetune the whole network during training except for the batch norm parameters of ResNet and the very early ResNet layers.

**Comment by LiXm1002** [comment]
Thanks you very much!

**Comment by LiXm1002** [comment]
I met another problem.T_T
I change resnet101 to resnet152, When training the loss value reduced from 9 to about 1. I think this is normal. But when I test it, there are nothing in the detections/23/0000.txt. Does anyone know why this is?

**Comment by ahnonay** [comment]
If you use another resnet backend the pretrained weights will be mostly useless. Maybe it helps to pretrain the Mask R-CNN parts of the model (basically everything except the association head and the conv3ds) on COCO and Mapillary, as we did for our pretrained model (the one with resnet101)

**Comment by LiXm1002** [comment]
Thanks！

**Comment by LiXm1002** [comment]
Then, when training the 101-layer pre-trained model do you still use the Track R-CNN or other networks?

**Comment by pvoigtlaender** [comment]
We pre-train on static Images (e.g. Mapillary Vistas), there you can just train as Mask R-CNN and add the re-id head when fine-tuning on the video data.

**Comment by LiXm1002** [comment]
I try to use the configs/normal_mask_rcnn to train a Mask R-CNN pre-train model, but I met the same problem if I use resnet-152 there are no output in forwarded. And I have a question.  Can I train a model without using a pre-trained model?

**Comment by ahnonay** [comment]
Unfortunately, it is hard to say why there are no outputs in forwarded/. You could try forwarding the training set and see if you get reasonable detections there.

You can train TrackR-CNN without a pre-trained model. However, the KITTI MOTS dataset is very small to train all parameters of such a huge model so I suspect that performance without pretraining will be very bad.

**Comment by LiXm1002** [comment]
Thanks for your suggestion.

**Comment by zhongzhuoyao** [comment]
Could you please show how to train a vanilla Mask R-CNN on COCO and Mapillary?

**Comment by ahnonay** [comment]
@zhongzhuoyao See https://github.com/VisualComputingInstitute/TrackR-CNN/issues/54
Closing this for now, please reopen if needed.
